src/Backend/SgateShopgatePlugin/Components/Cart.php

```diff
@@ -81,6 +89,14 @@ public function getCart($request, $httpResponse, $view)
             $this->session->offsetSet('promotionVouchers', $promotionVouchers);
         }
 
+        if (!empty($customerId) && $customerId !== "null") {
```

Comment:
Empty should already check for null:
https://www.php.net/manual/de/function.empty.php

Reply:
I need to check if the string is "null" so that's the reason for the second statement (user is guest). The first statement checks if the variable is empty because this variable is new and the value is empty in all older extension versions.